spark/src/main/kotlin/com/adevinta/spark/components/buttons/Button.kt

```diff
@@ -0,0 +1,193 @@
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.adevinta.spark.InternalSparkApi
+import com.adevinta.spark.PreviewTheme
+import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.components.buttons.SparkButton.TAG_PROGRESS_INDICATOR
+import com.adevinta.spark.components.icons.Icon
+import com.adevinta.spark.icons.SparkIcon
+import com.adevinta.spark.tools.modifiers.SlotArea
+import com.adevinta.spark.tools.modifiers.sparkUsageOverlay
+
+@InternalSparkApi
+@Composable
+fun SparkButton(
+    onClick: () -> Unit,
+    colors: ButtonColors,
+    contentPadding: PaddingValues,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    elevation: ButtonElevation? = ButtonDefaults.buttonElevation(),
+    border: BorderStroke? = null,
+    icon: SparkIcon? = null,
+    iconSide: IconSide = IconSide.START,
+    isLoading: Boolean = false,
+    content: @Composable RowScope.() -> Unit,
+) {
+    Button(
+        onClick = onClick,
+        modifier = modifier.sparkUsageOverlay(),
+        enabled = enabled,
+        elevation = elevation,
+        shape = SparkTheme.shapes.extraSmall,
+        border = border,
+        colors = colors,
+        contentPadding = contentPadding,
+    ) {
+        AnimatedVisibility(visible = isLoading) {
+            Row {
+                Box(Modifier.size(SparkButtonDefaults.IconSize)) {
+                    CircularProgressIndicator(
+                        Modifier.testTag(TAG_PROGRESS_INDICATOR),
+                        color = LocalContentColor.current,
+                        strokeWidth = 2.dp,
+                    )
+                }
+                Spacer(Modifier.width(8.dp))
+            }
+        }
+        if (icon != null && iconSide == IconSide.START) {
+            Icon(
+                sparkIcon = icon,
+                modifier = Modifier
+                    .size(SparkButtonDefaults.IconSize)
+                    .testTag("buttonIcon"),
+                contentDescription = null, // button text should be enough for context
+            )
+            Spacer(Modifier.width(SparkButtonDefaults.IconSpacing))
+        }
+        content()
+        if (icon != null && iconSide == IconSide.END) {
+            Spacer(Modifier.width(SparkButtonDefaults.IconSpacing))
+            Icon(
+                sparkIcon = icon,
+                modifier = Modifier
+                    .size(SparkButtonDefaults.IconSize)
+                    .testTag("buttonIcon"),
+                contentDescription = null, // button text should be enough for context
+            )
+        }
+    }
+}
+
+object SparkButton {
+    const val TAG_PROGRESS_INDICATOR = "progress_indicator"
+}
```

Comment:
Is this something that should be documented, and is it a convention that we want to use for other composables?
Using SparkButton object feels wrong to me 😬